bin/find-deprecated-objects.rb

```diff
@@ -24,17 +26,68 @@ def main
 
   namespaces = namespace_details
 
-  deprecated_api_objects.each do |obj|
-    puts object_csv(obj, namespaces)
-  end
-
-  stdout, _, _ = Open3.capture3("kubectl get pods --all-namespaces -o json")
-  # stdout = File.read("pods.json")
+  kubectl_api_objects.each { |obj| puts object_csv(obj, namespaces) }
 
   # TODO filter for tiller < 2.16.3
-  JSON.parse(stdout).fetch("items")
+  tiller_pods = parsed_json_output("kubectl get pods --all-namespaces -o json").fetch("items", [])
     .filter { |pod| tiller?(pod) }
-    .map { |pod| puts tiller_csv(pod, namespaces) }
+
+  tiller_pods.map { |pod| puts tiller_csv(pod, namespaces) }
+  output_helm2_object_data(tiller_pods, namespaces)
+  output_helm3_object_data(namespaces)
+end
+
+def helm3_deprecated_objects(namespace)
+  cmd = pluto_command(namespace, "3")
+  parsed_json_output(cmd).fetch("items", [])
+end
+
+def output_helm3_object_data(namespaces)
+  namespaces.keys.each do |namespace|
+    helm3_deprecated_objects(namespace)
+      .each { |object| puts helm_object_csv(object, namespaces) }
+  end
+end
+
+def output_helm2_object_data(tiller_pods, namespaces)
+  tiller_pod_namespaces = tiller_pods
+    .map { |pod| pod.dig("metadata", "namespace") }
+    .sort
+    .uniq
+
+  tiller_pod_namespaces.each do |ns|
+    helm2_deprecated_objects(ns).each do |obj|
+      puts helm_object_csv(obj, namespaces)
+    end
+  end
+end
+
+def helm_object_csv(object, namespaces)
+  team, repo = namespace_team_repo(object["namespace"], namespaces)
+
+  # 'name' is often something like "check-financial-eligibility/check-financial-eligibility"
+  # so if it matches that pattern, simplify it.
+  name = object.fetch("name")
+  /(.*)\/\1/.match(name) && name = $1
```

Comment:
@digitalronin , For now, it is good and shows the helm release name. But for later, to either display the whole name so we know the helm release name as well as resource name.